### ISSUE TITLE
Example application for React Native

### ISSUE THREAD

**Issue by petertonysmith94** [FuelLabs/fuels-ts#2570]
### Summary

Currently, we have no example applications that use React Native. The [documentation](https://reactnative.dev/docs/environment-setup), suggests to use [Expo](https://docs.expo.dev) so we should follow their advice:

- Create a React Native application in our `apps/*` directory:

  ```console
  npx create-expo-app@latest
  ```

- Add some simple functionality for testing purposes:
  - Create provider
  - Generate wallet
  - Check balance
  
- The following `Signer` logic has been reported to [fail](https://forum.fuel.network/t/fuel-ts-in-react-native/5954/8) during build, ~~it would be good to confirm~~ it is confirmed:

```ts
import { Signer } from "@fuel-ts/account";
const extendedPublicKey = Signer.extendPublicKey(new Uint8Array(publicKey));
```
---

# Scope of work for React Native support

## 1. Platform-specific Entry Points
- We’ll need React Native specific implementations alongside our current Node/browser paths
- Particularly for crypto operations (we’re heavily using Node’s crypto and browser’s Web Crypto API)
- Need to configure package.json with proper “react-native” entry points

## 2. Crypto Implementation
- Add React Native specific crypto implementations
- Will need to use RN-compatible alternatives like react-native-get-random-values
- May need polyfills for Buffer and other Node.js built-ins
- Test crypto operations thoroughly on mobile environment

## 3. Module Resolution
- Need to handle subpath exports properly for Metro bundler
- Currently seeing issues with imports like `@fuel-ts/address/configs`
- May need to restructure how we export our submodules

## 4. Documentation & Testing
- Add React Native setup instructions
- Document required polyfills and Metro config
- Set up CI with React Native test app
- Add React Native specific examples

## Questions to discuss:
- Do we want to maintain three separate implementations (Node/Browser/RN)?

**Comment by danielbate** [comment]
This could be a contender for `good-first-issue`? We've got some similar apps and its isolated from src, could be good for someone with native experience to pick up.

**Comment by YaTut1901** [comment]
hello, I can take this

**Comment by YaTut1901** [comment]
@petertonysmith94 hello, I have tried a couple of times to create app using expo, but it seems too complicated using pnpm and monorepo. Can I check other starters for react-native?

**Comment by YaTut1901** [comment]
@petertonysmith94 hello, regarding this task, I had several issues to build react-native with pnpm monorepo - unexpectedly that appeared to be not trivial. Now I can`t resolve the problem that fuels package that is decalared with "workspace:*" can`t be found when build with expo. Could you check if I miss something? The branch is here: https://github.com/YaTut1901/fuels-ts/tree/YaTut1901/feat/demo-react-native

**Comment by YaTut1901** [comment]
@petertonysmith94 hello, have you managed to check it?

**Comment by petertonysmith94** [comment]
Hey @YaTut1901 👋🏼 

I had a quick look, have you looked over the follow [resource](https://docs.expo.dev/guides/monorepos/) about managing a monorepo with Expo?

**Comment by YaTut1901** [comment]
@petertonysmith94 yes, I`ve done it accordingly with that page, but it can`t build project properly. So I`ve done it using workaround from this issue: https://github.com/expo/expo/issues/28775
Seems like working well unless it can`t resolve fuels package while building. Strange thing that it can resolve it while compiling and VS code show no warnings regards that

**Comment by petertonysmith94** [comment]
@YaTut1901 I haven't had the time to properly look over your solution yet. 

Potentially a better way forward, might be create a React Native app outside of our mono repo and getting that integrated with `fuels` successfully. That way, we can eliminate all the mono repo related issues (which I'm sure there are a few).

**Comment by YaTut1901** [comment]
@petertonysmith94 how do you suggest to do it? A separate repo?

**Comment by petertonysmith94** [comment]
@YaTut1901 Yeah a seperate repo would be ideal.

**Comment by BANG88** [comment]
hello Is there any progress? thank you.

**Comment by petertonysmith94** [comment]
Hey @BANG88 👋🏼 

No progress yet.
